File: Cola.py
class Cola:
    def __init__(self, n):
        self.arreglo = [0] * n
        self.inicio = 0
        self.fin = -1
        self.n = n
        self.numero_elementos = 0

    def insert(self, e):
        if self.isFull():
            print("La cola está llena")
        else:
            self.fin = (self.fin + 1) % self.n
            self.arreglo[self.fin] = e
            self.numero_elementos += 1

    def remove(self):
        if self.isEmpty():
            print("La cola está vacía")
            return None
        else:
            e = self.arreglo[self.inicio]
            self.inicio = (self.inicio + 1) % self.n
            self.numero_elementos -= 1
            return e

    def peek(self):
        if self.isEmpty():
            print("La cola está vacía")
            return None
        else:
            return self.arreglo[self.inicio]

    def isEmpty(self):
        return self.numero_elementos == 0

    def isFull(self):
        return self.numero_elementos == self.n

    def size(self):
        return self.numero_elementos


# Ejemplo de uso
if __name__ == "__main__":
    cola = Cola(5)
    cola.insert(10)
    cola.insert(20)
    print("Elemento en la parte delantera:", cola.peek())
    print("Eliminar elemento:", cola.remove())
    print("Elemento en la parte delantera:", cola.peek())

File: Cola.java
package practica3;

public class Cola {
    private long[] arreglo;
    private int inicio;
    private int fin;
    private int n;
    private int numeroElementos;

    public Cola(int n) {
        this.arreglo = new long[n];
        this.inicio = 0;
        this.fin = -1;
        this.n = n;
        this.numeroElementos = 0;
    }

    public void insert(long e) {
        if (isFull()) {
            System.out.println("La cola está llena");
        } else {
            fin = (fin + 1) % n;
            arreglo[fin] = e;
            numeroElementos++;
        }
    }

    public long remove() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
            return -1; // Valor de retorno indicativo de error
        } else {
            long e = arreglo[inicio];
            inicio = (inicio + 1) % n;
            numeroElementos--;
            return e;
        }
    }

    public long peek() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
            return -1; // Valor de retorno indicativo de error
        } else {
            return arreglo[inicio];
        }
    }

    public boolean isEmpty() {
        return numeroElementos == 0;
    }

    public boolean isFull() {
        return numeroElementos == n;
    }

    public int size() {
        return numeroElementos;
    }

    // Ejemplo de uso
    public static void main(String[] args) {
        Cola cola = new Cola(5);
        cola.insert(10);
        cola.insert(20);
        System.out.println("Elemento en la parte delantera: " + cola.peek());
        System.out.println("Eliminar elemento: " + cola.remove());
        System.out.println("Elemento en la parte delantera: " + cola.peek());
    }
}